### PR TITLE
cpu/samd21: fix vectors definition

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Some SAMD21 CPUs from the D series have an additional TCC3 and / or an additional analog comperator.

Add those to `vectors.c`.

Also rename the i2s interrupt to the proper name used by other sam0 devices.

### Testing procedure

No functional changes.

### Issues/PRs references

split off #14662

### CODE REVIEW

File: cpu/samd21/vectors.c [hunk 1]
```diff
@@ -66,8 +69,12 @@ ISR_VECTOR(1) const isr_t vector_cpu[CPU_IRQ_NUMOF] = {
     isr_rtc,                /*  3 Real-Time Counter */
     isr_eic,                /*  4 External Interrupt Controller */
     isr_nvmctrl,            /*  5 Non-Volatile Memory Controller */
+#ifdef REV_DMAC
     isr_dmac,               /*  6 Direct Memory Access Controller */
```
Comment (basilfx) [inline cpu/samd21/vectors.c:73]: Will this work with the `#ifdef`? If `REV_DMAC` is undefined, than the next item takes the position of `isr_dmac`.

Comment (basilfx) [reply cpu/samd21/vectors.c:73]: Same for the others.

Comment (benpicco) [reply cpu/samd21/vectors.c:73]: Yes that's the idea.

| samd20  | samd21 |
| ------------- | ------------- |
| image | image |

**Review by maribu — APPROVED** [review]
ACK. Vectors look identical to the datasheet tables quoted in https://github.com/RIOT-OS/RIOT/pull/15303/files#r512292975

**Pull request merged**